src/util/Util.js

```diff
@@ -22,17 +22,16 @@ class Util {
     if (text.length <= maxLength) return text;
     const splitText = text.split(char);
     if (splitText.length === 1) throw new RangeError('SPLIT_MAX_LEN');
-    const messages = [''];
-    let msg = 0;
-    for (let i = 0; i < splitText.length; i++) {
-      if (messages[msg].length + splitText[i].length + 1 > maxLength) {
-        messages[msg] += append;
-        messages.push(prepend);
-        msg++;
+    const messages = [];
+    let msg = '';
+    for (const chunk of splitText) {
+      if ((msg + char + chunk + append).length > maxLength) {
+        messages.push(msg + append);
+        msg = prepend;
       }
-      messages[msg] += (messages[msg].length > 0 && messages[msg] !== prepend ? char : '') + splitText[i];
+      msg += (msg && msg !== prepend ? char : '') + chunk;
     }
-    return messages.filter(m => m);
+    return messages.concat(msg).filter(m => m);
```

Comment:
messages should be an array, and msg should be a string. So I don't think you want Array.prototype.concat() there, as that's for joining 2 arrays, not an array and a string. You probably want push, like you have on line 29. Additionally I am pretty sure you want to add on the append there. (like the last message of a split codeblock)

Reply:
If you attempt to `.concat()` an `item` to an array that is not itself an array, it is transformed to `[item]`.

![image](https://user-images.githubusercontent.com/14080148/34938100-bd44a422-f9ef-11e7-8b94-28b11b1fa086.png)

I could use push, but I would need to do that on an extra line above, since push does not return the array but its length.

And no, I don't want to append to the last block:
![image](https://user-images.githubusercontent.com/14080148/34938207-12f3a62a-f9f0-11e7-8afd-3cbf25d12cde.png)

Reply:
Whoops, so you are right...